internal/host/transition_test.go

```diff
@@ -5988,3 +5989,153 @@ func validateEqualProgress(p1, p2 *models.HostProgressInfo) {
 		Expect(time.Time(p1.StageUpdatedAt).Equal(time.Time(p2.StageUpdatedAt))).To(BeTrue())
 	}
 }
+
+type testState struct {
+	state string
+}
+
+func newTestState(state string) *testState {
+	return &testState{
+		state: state,
+	}
+}
+
+func (ts *testState) State() stateswitch.State {
+	return stateswitch.State(ts.state)
+}
+
+func (ts *testState) SetState(state stateswitch.State) error {
+	ts.state = string(state)
+	return nil
+}
+
+var _ = Describe("State machine test", func() {
+	var (
+		ctx                   = context.Background()
+		mockController        *gomock.Controller
+		mockTransitionHandler *MockTransitionHandler
+		stateMachine          stateswitch.StateMachine
+		refreshHostArgs       TransitionArgsRefreshHost
+		testState             *testState
+	)
+
+	BeforeEach(func() {
+		mockController = gomock.NewController(GinkgoT())
+
+		mockTransitionHandler = NewMockTransitionHandler(mockController)
+
+		mockTransitionHandler.EXPECT().HasStatusTimedOut(gomock.Any()).Return(func(_ stateswitch.StateSwitch, _ stateswitch.TransitionArgs) (bool, error) {
```

Comment:
isn't that specific for your tests? maybe it should be under the same context or something like that.

Reply:
No, this function gets immediately called during `NewHostStateMachine`

Reply:
If this will cause problems to anyone in the future they can change it, this is not set in stone